\documentclass[10pt]{beamer}
\usetheme{Warsaw}
\usepackage{graphicx}
\usepackage[utf8]{inputenc}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage[notocbib]{apacite}
\usepackage{epstopdf}
\usepackage{booktabs}
\usepackage{colortbl, xcolor}

\setbeamertemplate{caption}{\centering\insertcaption\par}
\setlength{\belowcaptionskip}{15pt}
\renewcommand{\thetable}{}

\providecommand{\e}[1]{\ensuremath{\times 10^{#1}}}

\begin{document}


\date{}
\author{Micha\l{}  Burdukiewicz, Piotr Sobczyk}
\institute{ 
University of Wroc\l{}aw, Department of Genomics, Poland}

\title{Expanding signalHsmm using n-grams}


\begin{frame}
\maketitle

<<echo = FALSE,message=FALSE,results='asis'>>=
library(reshape2)
library(ggplot2)
library(grid)
library(biogram)
library(xtable)
library(seqinr)
library(signalHsmm)

theme_presentation <- theme(plot.background=element_rect(fill = "transparent",
                                                   colour = "transparent"),
                      plot.margin = unit(c(1,1,1,1), "cm"),
                      axis.text.x = element_text(size=7), 
                      axis.text.y = element_text(size=7),
                      axis.title.x = element_text(size=11, vjust = -1), 
                      axis.title.y = element_text(size=11, vjust = 1),
                      strip.text = element_text(size=8),
                      legend.text = element_text(size=7), 
                      legend.title = element_text(size=11),
                      plot.title = element_text(size=17),
                      panel.grid.major = element_line(colour="lightgrey", linetype = "dashed"),
                      panel.background = element_rect(fill = "transparent",colour = "black"),
                      legend.background = element_rect(fill="NA"))

@



\end{frame}

\begin{frame}
\frametitle{Outline}
\tableofcontents
\end{frame}

\section{Motivation}


\begin{frame}
To improve our model, we added few supplementary states representing specific motives that might occur in the proximity of cleavage site. The structure of cleavage sites, more conserved than other parts of signal peptide, may be reflected by n-grams (k-mers), short vectors of $n$ characters derived from input sequences.
\end{frame}

\begin{frame}
\begin{figure}[ht]\centering
\includegraphics[scale=0.42]{HSMMs}
\caption{The diagram of simple (A) and extended version of signalHsmm with the n-gram cleavage site model (B).}
\label{fig:enccomp}
\end{figure}

\end{frame}


\AtBeginSection[]
{
\begin{frame}<beamer>
\frametitle{Outline}
\tableofcontents[currentsection]
\end{frame}
}

\section{Impact of the n-gram extension}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis', fig.height=5.8>>=
load("ngram_extension.RData")

levels(plot_dat[["sig"]]) <- c("No", "Yes")

ggplot(plot_dat, aes(x=noMer,y=kMer, color=sig)) +
  geom_point() + 
  xlab("Normal model") + 
  ylab("Extended model") + 
  scale_color_discrete("Signal peptide") +
  geom_abline(slope=1) +
  theme_presentation +
  theme(legend.position="bottom")
@
\end{frame}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis'>>=
rws <- seq(1, 8 - 1, by = 2)
col <- rep("\\rowcolor[gray]{0.85}", length(rws))
print(xtable(bench_metrics2[1L:8, c("AUC", "TP", "TN", "FP", "FN")], "Performance of different classifiers.", digits = 4), include.rownames = TRUE, booktabs = TRUE,
      add.to.row = list(pos = as.list(rws), command = col), sanitize.text.function = identity,
      caption.placement = "top")
@
\end{frame}

\section{Differences in cleavage site composition}

\begin{frame}
\begin{enumerate}
\item Train signalHsmm on all sequences with signal peptide.
\item Test signalHsmm on training set.
\item Assign label "not recognized" to not recognized sequences.
\end{enumerate}
\end{frame}


\begin{frame}

Considering 2525 sequences, 149 (5.9\%) are wrongly recognized.

\end{frame}

\subsection{Amino acid sequences}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis', fig.height=5.8>>=
load("sumpres.RData")

ggplot(cs_unigram, aes(x = pos, y = freq, fill = rec)) +
  geom_bar(position = "dodge", stat = "identity") +
  facet_wrap(~aa, ncol = 5) + 
  scale_x_discrete("Position") +
  scale_y_continuous("Frequency") + 
  scale_fill_discrete("Identified") +
  theme_presentation +
  theme(legend.position="bottom")
@
\end{frame}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis', fig.height=5.8>>=
load("sumpres.RData")

ggplot(cs_bigram0, aes(x = pos, y = freq, fill = rec)) +
  geom_bar(position = "dodge", stat = "identity") +
  facet_wrap(~aa, ncol = 5) + 
  scale_x_discrete("Position") +
  scale_y_continuous("Frequency") + 
  scale_fill_discrete("Identified") +
  theme_presentation +
  theme(legend.position="bottom") +
  ggtitle("2-grams, distance 0")
@
\end{frame}



\begin{frame}
<<echo = FALSE,message=FALSE,results='asis', fig.height=5.8>>=
ggplot(cs_bigram1, aes(x = pos, y = freq, fill = rec)) +
  geom_bar(position = "dodge", stat = "identity") +
  facet_wrap(~aa, ncol = 5) + 
  scale_x_discrete("Position") +
  scale_y_continuous("Frequency") + 
  scale_fill_discrete("Identified") +
  theme_presentation +
  theme(legend.position="bottom") +
  ggtitle("2-grams, distance 1")
@
\end{frame}

\subsection{Degenerated amino acid sequences}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis'>>=
group2df <- function(group_list)
  data.frame(Groups = sapply(group_list, function(i)
    paste0(toupper(sort(i)), collapse = ", ")))

print(xtable(group2df(aaaggregation),
             caption = "Classification of amino acids used by signalHsmm."),
      include.rownames = FALSE)

@
\end{frame}


\begin{frame}
<<echo = FALSE,message=FALSE,results='asis', fig.height=5.8>>=
ggplot(cs_bigram0d, aes(x = pos, y = freq, fill = rec)) +
  geom_bar(position = "dodge", stat = "identity") +
  facet_wrap(~aa, ncol = 5) + 
  scale_x_discrete("Position") +
  scale_y_continuous("Frequency") + 
  scale_fill_discrete("Identified") +
  theme_presentation +
  theme(legend.position="bottom") +
  ggtitle("2-grams (degenerated), distance 0")
@
\end{frame}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis', fig.height=5.8>>=
ggplot(cs_bigram1d, aes(x = pos, y = freq, fill = rec)) +
  geom_bar(position = "dodge", stat = "identity") +
  facet_wrap(~aa, ncol = 5) + 
  scale_x_discrete("Position") +
  scale_y_continuous("Frequency") + 
  scale_fill_discrete("Identified") +
  theme_presentation +
  theme(legend.position="bottom") +
  ggtitle("2-grams (degenerated), distance 1")
@
\end{frame}

\section{Heuristic algorithm}

\begin{frame}
The heuristic algorithm relies on presence of some amino acids to define borders of regions.

According to the origingal authors, C-region is recognized by a presence of: R, H, K, D, E.
\end{frame}

\begin{frame}
\begin{figure}[ht]\centering
\includegraphics[scale=0.42]{enccomp}
\end{figure}
\end{frame}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis'>>=
print(xtable(group2df(aaaggregation),
             caption = "Classification of amino acids used by signalHsmm."),
      include.rownames = FALSE)
@
\end{frame}

\begin{frame}

After addition of G, S, T, P, Y, 135 (5.3\%) sequences are wrongly recognized (compared to 149).

\end{frame}

\section{Summary table}

\begin{frame}
<<echo = FALSE,message=FALSE,results='asis'>>=
rws <- seq(1, nrow(bench_metrics2) - 1, by = 2)
col <- rep("\\rowcolor[gray]{0.85}", length(rws))
print(xtable(bench_metrics2[, c("AUC", "TP", "TN", "FP", "FN")], "Performance of different classifiers.", digits = 4), include.rownames = TRUE, booktabs = TRUE,
      add.to.row = list(pos = as.list(rws), command = col), sanitize.text.function = identity,
      caption.placement = "top")
@
\end{frame}

\end{document}